### DISCUSSION TITLE
[Feature] Duplicate detection improvements.

### DISCUSSION THREAD

**Discussion by Dental1347** [immich-app/immich#14297] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hello,

As of recently I have been using the duplicate detection tool built into immich. It has been working great, but it seems to have very basic functionalities at the moment. It seems to look at filesize mostly it seems.

I would like to suggest some features added, I searched around but couldn't find a FR for specific improvements, perhaps those can be introduced here? 

A few of mine:
1. Add the support to detect duplicate video's. Currently it seems to only detect images (I have not tested if it works for the apple motion images). Having this tool work on videos would be absolutely fantastic!
2. Possibility (ties in with perhaps the workflows) to add preferences to what you would like to keep if duplicates are found. Let's say you prefer a certain image format over another (HEIC vs JPG, as an example). These will make the removal process faster.
3. Take metadata of the files into account. I don't know if this is already being done, but something like this would also be nice to quickly distinguish the original video/photo from let's say a whatsapp photo/video where all metadata is stripped.

These are a few of my suggestions that would make the tool a lot more solid. Would like to hear your opinions. Hope a FR like this one doesn't already exist, I have looked for it build did not find it.

Kind regards.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Like the template requests, please stick to only one FR per post. Also, 2 and 3 are duplicates.

**Reply by Dental1347** [reply]
What post is this a duplicate of? Would be nice to know :)
